### PR TITLE
TF `try_gpu` string output fix

### PR DESCRIPTION
**Pull request by biswajitsahoo1111**
*Description of changes:*

This pull request only affects the string output that is generated after training `seq2seq` model. For example, the following old output
```
x tokens/sec on <tensorflow.python.eager.context._EagerDeviceContext object at 0x7ff8005deb00>
```
becomes
```
x tokens/sec on /GPU:0
```

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

File: d2l/tensorflow.py [hunk 1]
```diff
@@ -14,12 +14,13 @@
 import time
 import zipfile
 from collections import defaultdict
+
+import numpy as np
```
Comment (AnirudhDagar) [inline d2l/tensorflow.py:18]: Please revert these changes.

File: d2l/tensorflow.py [hunk 2]
```diff
@@ -14,12 +14,13 @@
...
 import pandas as pd
 import requests
 from IPython import display
 from matplotlib import pyplot as plt
 
-import numpy as np
```
Comment (AnirudhDagar) [inline d2l/tensorflow.py:22]: Please revert these changes.

File: d2l/tensorflow.py [hunk 3]
```diff
@@ -43,6 +44,7 @@
 import time
 import zipfile
 from collections import defaultdict
+
```
Comment (AnirudhDagar) [inline d2l/tensorflow.py:47]: Please remove this addition of blank line.

File: d2l/tensorflow.py [hunk 4]
```diff
@@ -51,8 +53,10 @@
 d2l = sys.modules[__name__]
 
 import numpy as np
+
 import tensorflow as tf
 
+
 def use_svg_display():
```
Comment (AnirudhDagar) [inline d2l/tensorflow.py:60]: Please revert these changes.

File: d2l/tensorflow.py [hunk 5]
```diff
@@ -1847,7 +1851,7 @@ def train_seq2seq(net, data_iter, lr, num_epochs, tgt_vocab, device):
         if (epoch + 1) % 10 == 0:
             animator.add(epoch + 1, (metric[0] / metric[1],))
     print(f'loss {metric[0] / metric[1]:.3f}, {metric[1] / timer.stop():.1f} '
-          f'tokens/sec on {str(device)}')
+          f'tokens/sec on {str(device._device_name)}')
```
Comment (AnirudhDagar) [inline d2l/tensorflow.py:1854]: Please make the same change in https://github.com/d2l-ai/d2l-en/blob/master/chapter_appendix-tools-for-deep-learning/utils.md and then run d2lbook build lib. We don't directly change the lib files, these are automatically generated using the d2lbook package after editing the source files where these functions are saved.

**Review by AnirudhDagar — CHANGES_REQUESTED** [review]
This works and definitely shows a much better output for device name, but I'm not sure using a private attribute is the best thing to do here. I don't think there is any other way in TF to achieve this. Thoughts @astonzhang?

@biswajitsahoo1111 thanks for your contribution, I've made some comments, please address those. You will need the d2lbook package generally for d2l development. Feel free to refer to CONTRIBUTING.md for more info :))

**Comment by biswajitsahoo1111** [comment]
running `d2lbook build lib` has significantly changed all files in `d2l` folder. Don't know why. Looking into it.

**Comment by AnirudhDagar** [comment]
Can you please make sure you have the most recent version of d2lbook and you run it after rebasing your branch with the latest changes in master?

`pip install --upgrade git+https://github.com/d2l-ai/d2l-book`

**Pull request merged**